index.test-d.ts

```diff
@@ -0,0 +1,20 @@
+import { expectType, expectAssignable } from 'tsd'
+import * as closeWithGrace from '.'
+import { Options, CloseWithGraceCallback, Signals } from '.'
+
+expectAssignable<CloseWithGraceCallback>((param: { err: Error, manual: boolean, signal: Signals }, cb) => Promise.resolve())
```

Comment:
this should not be possible.

Reply:
I changed the tests at least to match with the correct expectations is this what you had in mind ? 

```ts
expectAssignable<CloseWithGraceCallback>(async (param: { err: Error, manual: boolean, signal: Signals }) => { })
expectAssignable<CloseWithGraceCallback>((param: { signal: Signals }, cb) => { })
```

also I did some more investigation no luck that was the closer I got but couldn't port it to the `.d.ts`
[TS playground](KYDwDg9gTgLgBDAnmYcDCAjOBeOAKYKKaAfgC44BRI6AShwD44A3CASwBMBuAKFElgJkqAAoBDKGIC2AZxxwA3jzgq4hKOSo0ovVXCliAdgFcxAG00YIEM8CO7VMtgHND5zQHIAygEkA4gAqlABKALIecAA)